src/ArrayHelper.php

```diff
@@ -428,20 +432,45 @@ public static function setValueByPath(array &$array, $path, $value, string $deli
 
     /**
      * @param array|float|int|string $path The path of where do you want to write a value to `$array`.
-     * The path can be described by a string when each key should be separated by delimiter.
+     * The path can be described by a string when each key should be separated by delimiter. If a path item contains
+     * delimiter, it can be escaped with "\" (backslash) or a custom delimiter can be used.
      * You can also describe the path as an array of keys.
-     * @param string $delimiter A separator, used to parse string $key for embedded object property retrieving. Defaults
+     * @param string $delimiter A separator, used to parse string key for embedded object property retrieving. Defaults
      * to "." (dot).
+     * @param bool $escapeDelimiter Whether to escape delimiter in the items of final array (in case of using string as
+     * an input) When `true`, "\" (backslashes) are removed - "\." will become "." for "." delimiter. Defaults to
+     * `false meaning backslashes will persist.
```

Comment:
```suggestion
     * @param bool $escapeDelimiter Whether to escape delimiter in the items of final array (in case of using string as
     * an input). When `false`, "\" (backslashes) are removed. For a "." as delimiter, "\." becomes ".". Defaults to `false`.
```